src/test/java/edu/harvard/iq/dataverse/util/FileUtilTest.java

```diff
@@ -61,6 +61,49 @@ public void testGetCiteDataFileFilename() {
         }
     }
 
+    @RunWith(Parameterized.class)
+    public static class FileUtilParamTest2 {
+
+        @Parameters
+        public static Collection data() {
+            return Arrays.asList(new Object[][] {
+                // functional approach: what should the method do
+                // replace no extension with an empty extension
+                { "no-extension", "no-extension", ""},
+
+                // replace extension x with same extension
+                { "extension.x", "extension.x", "x" },
+
+                // replace extension x with another extension y
+                { "extension.y", "extension.x", "y" },
+
+                // interface approach: what are possible inputs
+                { null, null, null },
+                { null, null, "" },
+                { null, null, "y" },
+
+                { "", "", null },
+                { "", "", "" },
+                { "", "", "y" },
+            });
+        }
+
+        @Parameter
```

Comment:
could you move these parameter definitions before the Parameters list? its a bit more readable.

Reply:
Sure, I made the change in 2c612b3